### DISCUSSION TITLE
Modeling interactions in the model rather than the usecase

### DISCUSSION THREAD

**Discussion by jondavid-black** [DevOps-MBSE/AaC#180] [Ideas]
AaC focuses on clearly modeling the structure of a system and the interactions among the parts of the system to perform desired behaviors.  I've been white boarding some UI thoughts around how to visually represent these 2 aspects.  The AaC spec does a good (simple/clean) job of capturing structure within the `model` item using `components`.  But to really understand interactions there's quite a bit of inference to be done, and to really understand things you have to incorporate the `usecase` into the thought process.

I propose removing `usecase` as a separate root type and integrating it into the `behavior` element of a `model`.  I think all we'd really need is the `steps` field because we already have `components` defined for the `model` which should parallel the `participants`.  There is a risk this may get a little confusing along side of the `acceptance` field so we need to consider it carefully.  That said, I think it drives better clarity overall by more closely integrating interaction information into the structure so that it is captured within a useful context.  If we do this, including `steps` would be optional and would only make sense when a `model` has `components` that interact to perform a `behavior`.

There would be some breakage to account for in the PlantUML plugin that generated use case diagrams if we did this.  I'm not aware of anything else that would break.  It may be worth doing an experiment to validate that.

**Comment by Coffee2Bits** [comment]
I'm really behind the idea of tying usecases more closely to the models, as I mentioned when talking about disliking how the 'root' file changes when you want to say look at a system model vs a system's interactions. 

```I think all we'd really need is the steps field because we already have components defined for the model which should parallel the participants.```

Wouldn't this limit usecases to interactions amongst the components of a model? For instance, if we had models external to a system that interact with the system how would we incorporate those external models _which aren't components of the system_?

**Reply by Coffee2Bits** [reply]
That being said, using `.aac` as a file extension would make our lives a little easier when it comes to the VSCode Plugins since they can key-off the file extension to determine language types.
